src/uv-common.c

```diff
@@ -155,6 +155,22 @@ static const char* uv__unknown_err_code(int err) {
   return copy != NULL ? copy : "Unknown system error";
 }
 
+#define UV_ERR_NAME_GEN_R(name, _) case UV_ ## name: { strncpy(buf, #name, buflen); break; }
+const char* uv_err_name_r(int err, char *buf, size_t buflen) {
```

Comment:
Style: `char* buf`

Substance: since this function returns `buf`, its return value should arguably be `char*`, not `const char*`.

Aside: you mention `strerror_r()`, but that function returns an int, not a pointer.